edn/src/edn.rustpeg

```diff
@@ -0,0 +1,106 @@
+
+use std::collections::{BTreeSet, BTreeMap, LinkedList};
+use std::iter::FromIterator;
+use types::{Pair, Value};
+use ordered_float::OrderedFloat;
+
+// Goal: Be able to parse https://github.com/edn-format/edn
+// Also extensible to help parse http://docs.datomic.com/query.html
+
+// TODO: Support tagged elements
+// TODO: Support comments
+// TODO: Support discard
+
+#[export]
+nil -> Value = "nil" {
+    Value::Nil
+}
+
+#[export]
+boolean -> Value =
+    "true" { Value::Boolean(true) } /
+    "false" { Value::Boolean(false) }
+
+digit = [0-9]
+sign = "-" / "+"
+
+#[export]
+integer -> Value = i:$( sign? digit+ ) {
+    Value::Integer(i.parse::<i32>().unwrap())
```

Comment:
I think `i32` is probably wrong. Clojure automatically walks the numeric tower:

```clojure
user=> {:foo 12345423857458454}
{:foo 12345423857458454}
user=> {:foo 123454238574584543434343434343}
{:foo 123454238574584543434343434343N}
```

but at the very least we will want to support microsecond timestamps, which are bigger than a 32-bit signed int.